### PR TITLE
[SPARK-6268][MLlib] KMeans parameter getter methods

### PR DESCRIPTION
**Pull request by hhbyyh**
jira: https://issues.apache.org/jira/browse/SPARK-6268

KMeans has many setters for parameters. It should have matching getters.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala [hunk 1]
```diff
@@ -78,6 +93,11 @@ class KMeans private (
   }
 
   /**
+   * Number of runs of the algorithm to execute in parallel.
+   */
+  def getRuns: Int = runs
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: `setRuns` is an experimental API, so it should be the same for `getRuns`.

**Comment by mengxr** [comment]
LGTM except one minor inline comment.

**Comment by hhbyyh** [comment]
@mengxr Thanks for review. New commit submitted.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala [hunk 2]
```diff
@@ -52,12 +52,22 @@ class KMeans private (
    */
   def this() = this(2, 20, 1, KMeans.K_MEANS_PARALLEL, 5, 1e-4, Utils.random.nextLong())
 
+  /**
+   * Number of clusters to create (k).
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: Nit: You could prefix this with `@return`
Interesting question: naming getters `getFoo` is of course standard in Java, and seems just fine too, although in Scala we'd be more used to accessing `foo` only. It's possible to declare `private var _foo` and then declare the public getter to have name `foo` here. Does anyone prefer that?

**Comment by hhbyyh** [comment]
Hi @srowen Thanks for the suggestions.

1. For adding the `@return` prefix. Sounds fair, and if we're doing it, shall we change other implementations as well? (like LDA, GaussianMixture and others).

2. For using foo rather than getFoo, again it may impact a wide range of code. And shall we change setters to `age_= (value:Int) : this.type` ? @srowen, @mengxr ,@jkbradley. I'm not sure what's the general recommendation within Spark. And I'd appreciate if there can be a quick decision from community leads. Thanks.

**Comment by srowen** [comment]
Ignore my second comment; it would take a fair bit more change including changing current setters. 
I would not change javadoc on other code, no. LGTM.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**